### DISCUSSION TITLE
Add FCPXML v1.12 Support

### DISCUSSION THREAD

**Discussion by latenitefilms** [orchetect/swift-daw-file-tools#29] [Ideas]
**Summary:**
- Added `<!ATTLIST filter-video nameOverride CDATA #IMPLIED>`
- Added `<!ATTLIST filter-audio nameOverride CDATA #IMPLIED>`
- Updated `<!ATTLIST conform-rate frameSampling (floor | nearest-neighbor | frame-blending | optical-flow-classic | optical-flow | optical-flow-frc) "floor">`
- Updated `<!ATTLIST timeMap frameSampling (floor | nearest-neighbor | frame-blending | optical-flow-classic | optical-flow | optical-flow-frc) "floor">`
- Updated `<!ATTLIST match-text rule (includes | doesNotInclude | is | isNot | startsWith | endsWith) "includes">`
- Updated `<!ATTLIST match-property rule (includes | doesNotInclude | is | isNot | isSet | isNotSet | startsWith | endsWith) "includes">`

**DTD:**
[FCPXMLv1_12.dtd.zip](https://github.com/user-attachments/files/15923884/FCPXMLv1_12.dtd.zip)

**Comment by orchetect** [comment]
- Version 1.12 has been added to `FinalCutPro.FCPXML.Version` and the type has been refactored from an enum to a struct so as to be able to support arbitrary (and future) FCPXML version numbers, with equality and comparability.
- `filter-video` is not yet abstracted as a model object in DAWFileKit so no changes are necessary.
- `filter-audio` is not yet abstracted as a model object in DAWFileKit so no changes are necessary.
- `FrameSampling` model object for both `ConformRate` and `TimeMap` gains the new `opticalFlowFRC` case
- `match-text` is not yet abstracted as a model object in DAWFileKit so no changes are necessary.
- `match-property ` is not yet abstracted as a model object in DAWFileKit so no changes are necessary.

These updates will be in the next release.
